AnimationGroup.lua

```diff
@@ -60,28 +60,49 @@ function CreateFrame(...)
     return frame
 end
 
-
 local function OnUpdate(self, elapsed)
     if self.paused then
         return
     end
-
-    self.time = self.time + (self.group.reverse and -elapsed or elapsed)
+    self.delaying = true
+
+    if not self.group.reverse and self.startdelay and self.startdelayTime < self.startdelay then
+        self.startdelayTime = self.startdelayTime + elapsed
+    elseif self.group.reverse and self.enddelay and self.enddelayTime < self.enddelay then
+        self.enddelayTime = self.enddelayTime + elapsed
+    elseif (not self.group.reverse and self.time < self.duration) or (self.group.reverse and self.time > 0) then
+        self.time = self.time + (self.group.reverse and -elapsed or elapsed)
+        self.delaying = false
+        self.progress = self.time / self.duration
+        self.progress = self.progress < 0 and 0 or self.progress
+        self.progress = self.progress > 1 and 1 or self.progress
```

Comment:
```suggestion
        self.progress = max(min(self.progress, 1), 0)
```

`max` and `min` are available globally iirc.

Reply:
calling of function much slower then just compare operator

Reply:
Calling global functions is much slower than compare

Reply:
True - you can make a local copy at init time (global to the file) - that's the normal pattern :)

```
local min = min
local max = max
```

Obviously, we need to be careful adding too much logic and expensive calculations to an OnUpdate function, such that we don't block the render thread too much.

Reply:
I know my code less expensive and i can show u tests:

![calcs](https://user-images.githubusercontent.com/20452393/195395186-3b0441f4-614c-4f17-9e5c-86d98d677ff5.PNG)

Reply:
How does it perform if you are to make your own local function, call it (e.g.) `BoundPercentage` with the alternative min/max logic. Does it suffer from the same method call penalty? If we can improve readability, that would be perfect. Otherwise a comment stating the behavior would suffice.

Reply:
Its less than min\max but also 2 times slower

![calcs2](https://user-images.githubusercontent.com/20452393/195402613-b39cb4ff-56bb-4667-86c0-660a92c26167.PNG)